pkg/main.go

```diff
@@ -5,15 +5,15 @@ import (
 
 	"github.com/grafana/athena-datasource/pkg/athena"
 	"github.com/grafana/athena-datasource/pkg/athena/routes"
+	"github.com/grafana/grafana-aws-sdk/pkg/awsds"
 	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
 	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
-	"github.com/grafana/sqlds/v2"
 )
 
 func main() {
 	// Start listening to requests sent from Grafana.
 	s := athena.New()
-	ds := sqlds.NewDatasource(s)
+	ds := awsds.NewAsyncAWSDatasource(s)
```

Comment:
very interesting! I haven't been following along that much with this work maybe at some point y'all could talk about this big refactor and learnings you had along the way? 

I have also run into the question of "is this a sql datasource that uses aws or is this an aws datasource that uses sql" and found the question frustrating... like maybe this is a datasource that has aws-like attributes and sql-like attributes and we want to mixin different util like things. 

None of this is really relevant to this pr of course just talking allow for future work possibilities.